Tests/Dbal/MutableAclProviderTest.php

```diff
@@ -261,10 +261,18 @@ public function testUpdateDoesNothingWhenThereAreNoChanges()
             ->expects($this->never())
             ->method('beginTransaction')
         ;
-        $con
-            ->expects($this->never())
-            ->method('executeUpdate')
-        ;
+        if (method_exists(Connection::class, 'executeUpdate')) {
+            // DBAL < 4.0
+            $con
+                ->expects($this->never())
+                ->method('executeUpdate')
+            ;
+        } else {
```

Comment:
What if both methods exist?

Reply:
As far as I can see in doctrine/dbal 3.9 and 3.10, both methods exist. However, `executeUpdate` is marked as deprecated, so it might be better to use `executeStatement` directly.